### PR TITLE
[Docs] Adding Quick start examples for Hibernate and spring-data-yugabytedb

### PR DESCRIPTION
**Pull request by nchandrappa**
Adding Quick start examples for `Hibernate` and `spring-data-yugabytedb` under Java build an application page.

### CODE REVIEW

File: docs/content/preview/quick-start/build-apps/java/ysql-hibernate.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-spring-data/" class="nav-link">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL - Spring Data JPA
+    </a>
+  </li>
+   <li>
+    <a href="../ysql-ebeans/" class="nav-link active">
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-hibernate.md:56]:
```suggestion
    <a href="../ysql-ebeans/" class="nav-link">
```

File: docs/content/preview/quick-start/build-apps/java/ysql-hibernate.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-jdbc-ssl/" class="nav-link">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL - JDBC SSL/TLS
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-hibernate/" class="nav-link">
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-hibernate.md:38]:
```suggestion
    <a href="../ysql-hibernate/" class="nav-link active">
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-hibernate/" class="nav-link">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL - Hibernate
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-sdyb/" class="nav-link">
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:44]:
```suggestion
    <a href="../ysql-sdyb/" class="nav-link active">
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    </a>
+  </li>
+  <li >
+    <a href="../ysql-spring-data/" class="nav-link">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL - Spring Data JPA
+    </a>
+  </li>
+   <li>
+    <a href="../ysql-ebeans/" class="nav-link active">
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:56]:
```suggestion
    <a href="../ysql-ebeans/" class="nav-link">
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+
+This tutorial assumes that:
+
+- YugabyteDB is up and running. If you are new to YugabyteDB, you can download, install, and have YugabyteDB up and running within five minutes by following the steps in [Quick start](../../../../quick-start/).
+- Java Development Kit (JDK) 1.8, or later, is installed. JDK installers for Linux and macOS can be downloaded from [OpenJDK](http://jdk.java.net/), [AdoptOpenJDK](https://adoptopenjdk.net/), or [Azul Systems](https://www.azul.com/downloads/zulu-community/).
+- [Apache Maven](https://maven.apache.org/index.html) 3.3 or later, is installed.
+
+## Create the Spring Data YugabyteDB app using Spring Boot
+
+Spring Boot project provides [Spring Initializr](https://start.spring.io/) utility for generating dependencies for Spring Boot applications.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:85]:
```suggestion
The Spring Boot project provides the [Spring Initializr](https://start.spring.io/) utility for generating dependencies for Spring Boot applications.
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+
+1. Navigate to [Spring Initializr](https://start.spring.io/). This service pulls in all the dependencies you need for an application and does most of the setup for you.
+
+1. Choose either Maven and Java programming language.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:89]: either and?

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+
+1. Click Dependencies and select `Spring Web` and `PostgreSQL Driver`. Click Generate.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:91]:
```suggestion
1. Click **Dependencies** and select `Spring Web` and `PostgreSQL Driver`. Click **Generate**.
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+
+1. Download the resulting ZIP file, which is an archive of Spring Boot application that is configured with your choices.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:93]:
```suggestion
1. Download the resulting ZIP file, which is an archive of a Spring Boot application that is configured with your choices.
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+
+1. Add the following dependencies to `pom.xml` of the Spring Boot application
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:95]:
```suggestion
1. Add the following dependencies to `pom.xml` of the Spring Boot application:
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+<dependency>
+  <groupId>com.zaxxer</groupId>
+  <artifactId>HikariCP</artifactId>
+</dependency>
+
+```
+
+## Create the sample Spring Data YugabyteDB application
+
+1. Create a new files `Employee.java`, `EmployeeRepository.java`, and `YsqlConfig.java` domain object in the base directory.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:112]: Are they files or domain objects? "Create a new files"?

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 11]
```diff
@@ -0,0 +1,318 @@
...
+    
+    @Bean
+    TransactionManager transactionManager(DataSource dataSource) {                     
+        return new YugabyteTransactionManager(dataSource);
+    }
+
+}
+```
+
+2. There are a number of options that can be customized in the properties file located at src/main/resources/application.properties. Given YSQL's compatibility with the PostgreSQL language, the spring.jpa.database property is set to POSTGRESQL and the spring.datasource.url is set to the YSQL JDBC URL: jdbc:postgresql://localhost:5433/yugabyte.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:224]:
```suggestion
2. A number of options can be customized in the properties file located at `src/main/resources/application.properties`. Given YSQL's compatibility with the PostgreSQL language, the `spring.jpa.database` property is set to POSTGRESQL and the `spring.datasource.url` is set to the YSQL JDBC URL: jdbc:postgresql://localhost:5433/yugabyte.
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+
+```java
+
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:227]: extra space

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 13]
```diff
@@ -0,0 +1,318 @@
...
+spring.datasource.username=yugabyte
+spring.datasource.password=
+
+# HikariCP config (pool size, default isolation level).
+spring.datasource.type=com.zaxxer.hikari.HikariDataSource
+spring.datasource.hikari.transactionIsolation=TRANSACTION_SERIALIZABLE
+
+```
+
+3. Create Spring Boot Application runner for performing read and write against YugabyteDB Cluster.
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:253]:
```suggestion
3. Create a Spring Boot Application runner to perform reads and writes against the YugabyteDB Cluster.
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 14]
```diff
@@ -0,0 +1,318 @@
...
+  
+  customerRepository.save(customer);
+  
+  Employee customerFromDB = null;
+  customerFromDB = customerRepository.findByEmail("user@one.com");
+  
+  System.out.println(String.format("Query returned: name = %s, email = %s",
+    customerFromDB.getName(), customerFromDB.getEmail()));
+  
+
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:299]: indenting and spacing of these last 2 brackets is wonky

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 15]
```diff
@@ -0,0 +1,318 @@
...
+}
+```
+
+## Run your new program
+
+```sh
+$ ./mvnw spring-boot:run
+```
+
+you should see the following as the output:
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:311]:
```suggestion
You should see the following output:
```

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md [hunk 16]
```diff
@@ -0,0 +1,318 @@
...
+
+```text
```
Comment (ddhodge) [inline docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:313]:
```suggestion
```output
```

**Review by ddhodge — APPROVED** [review]
Minor comments; buttons at top need small fix to on state

File: docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md (hunk 14 above)
Comment (nchandrappa) [reply docs/content/preview/quick-start/build-apps/java/ysql-sdyb.md:299]: fixed.

**Pull request merged**